api/src/main/java/org/openmrs/module/cfl/api/service/messages/impl/MultiProjectDefaultPatientTemplateServiceImpl.java

```diff
@@ -53,7 +63,12 @@ private List<PatientTemplate> getProjectBasedPatientTemplatesWithDefaultValues(P
         new ArrayList<>(templates.size() + templates.size() * actors.size());
 
     for (Template template : templates) {
-      patientTemplates.add((new PatientTemplateBuilder(template, patient)).build());
+      PatientTemplate patientTemplate =
+          Context.getService(PatientTemplateService.class)
+              .getOrCreatePatientTemplate(patient, template.getName());
+      if (patientTemplate.getId() == null) {
```

Comment:
How this logic works? 
`getOrCreatePatientTemplate` says get or create, meaning it will return existing or create new one. A valid null return value looks confusing now, hmm.

Reply:
I forgot to change name of called method. It has been changed to getOrBuildPatientTemplate. When patient template is not found then new patient template object is built (but not saved in database) that's why we check patientTemplate.**getId()** instead of patientTemplate